packages/popper/src/use-popper.ts

```diff
@@ -1,170 +1,155 @@
+import { Modifier } from "@popperjs/core"
+import { createElement, CSSProperties, useCallback, useState } from "react"
+import { usePopper as useBasePopper } from "react-popper"
+import { getBoxShadow, toTransformOrigin } from "./popper.utils"
 
 export type { Placement }
 
 export interface UsePopperProps {
   gutter?: number
   placement?: Placement
-  offset?: number
+  offset?: [number, number]
   preventOverflow?: boolean
   fixed?: boolean
-  forceUpdate?: boolean
   flip?: boolean
   arrowSize?: number
+  matchWidth?: boolean
   arrowShadowColor?: string
-  eventsEnabled?: boolean
-  modifiers?: Modifier<any, any>[]
+  modifiers?: Modifier<string, any>[]
 }
 
 export function usePopper(props: UsePopperProps = {}) {
   const {
-    placement: initialPlacement = "bottom",
-    offset: offsetProp,
+    placement = "bottom",
     preventOverflow = true,
     fixed = false,
-    forceUpdate = true,
     flip = true,
-    arrowSize = 10,
+    arrowSize = 8,
+    gutter = 8,
+    offset,
     arrowShadowColor,
-    gutter = arrowSize,
-    eventsEnabled = true,
+    matchWidth,
     modifiers,
   } = props
 
-  const popper = React.useRef<Instance | null>(null)
-  const referenceRef = React.useRef<HTMLButtonElement>(null)
-  const popoverRef = React.useRef<HTMLDivElement>(null)
-  const arrowRef = React.useRef<HTMLDivElement>(null)
-
-  const [originalPlacement, place] = React.useState(initialPlacement)
-  const [placement, setPlacement] = React.useState(initialPlacement)
-  const [offset] = React.useState(offsetProp || [0, gutter])
-  const [popoverStyles, setPopoverStyles] = React.useState<React.CSSProperties>(
-    {},
+  const [referenceNode, setReferenceNode] = useState<HTMLButtonElement | null>(
+    null,
   )
-  const [arrowStyles, setArrowStyles] = React.useState<React.CSSProperties>({})
+  const [popperNode, setPopperNode] = useState<HTMLDivElement | null>(null)
+  const [arrowNode, setArrowNode] = useState<HTMLDivElement | null>(null)
 
-  const update = React.useCallback(() => {
-    if (popper.current) {
-      popper.current.forceUpdate()
-      return true
-    }
-    return false
-  }, [])
+  const popperJS = useBasePopper(referenceNode, popperNode, {
+    placement,
+    strategy: fixed ? "fixed" : "absolute",
+    modifiers: [
+      {
+        name: "offset",
+        options: {
+          offset: offset ?? [0, gutter],
+        },
+      },
+      {
+        name: "flip",
+        enabled: flip,
+        options: { padding: 8 },
+      },
+      {
+        name: "arrow",
+        enabled: !!arrowNode,
+        options: { element: arrowNode },
+      },
+      // {
+      //   name: "computeStyles",
+      //   options: { gpuAcceleration: false },
+      // },
+      {
+        name: "preventOverflow",
+        enabled: !!preventOverflow,
+      },
+      {
+        name: "matchWidth",
+        enabled: !!matchWidth,
+        phase: "beforeWrite",
+        requires: ["computeStyles"],
+        fn: useCallback(({ state }) => {
```

Comment:
this useCallback as well as the one below should most likely be declared outside of `useBasePopper`

considering the generally huge object, would it makse sense to do something like this:
```js
const popperJS = useBasePopper(referenceNode, popperNode, getConfig({
  placement,
  fixed,
  offset,
  gutter,
  arrowNode,
  preventOverflow,
  /* etc */
})
```
then we could test that function separately, too, because it oculd live outside of the hook?

Reply:
Yes, I thought about this as well. I was indifferent about it considering we might not be testing that function in isolation.